From 3e266273f2b1483ca0fdecbd4b06c45677922169 Mon Sep 17 00:00:00 2001
From: Lubomir Rintel <lkundrak@v3.sk>
Date: Sat, 29 Oct 2016 13:25:49 +0200
Subject: [PATCH] msg_init: the memory size is 4 octets

Without this the memory size would be clamped to zero for >64k devices.
This caused SST28SF040A not ot be erased completely. I guess most chips
such as microcontrollers don't really care about the size since they can
be erased all at once.
---
 minipro.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/minipro.c b/minipro.c
index e6fc9a8..cdd1f6b 100644
--- a/minipro.c
+++ b/minipro.c
@@ -54,7 +54,7 @@ static void msg_init(unsigned char *out_buf, unsigned char cmd, device_t *device
 	format_int(&(out_buf[9]), device->opts3, 2, MP_LITTLE_ENDIAN);
 
 	out_buf[11] = icsp;
-	format_int(&(out_buf[12]), device->code_memory_size, 2, MP_LITTLE_ENDIAN);
+	format_int(&(out_buf[12]), device->code_memory_size, 4, MP_LITTLE_ENDIAN);
 }
 
 static unsigned int msg_transfer(minipro_handle_t *handle, unsigned char *buf, int length, int direction) {
